src/sanitizer/rds.py

```diff
@@ -0,0 +1,329 @@
+
+
+def wait_resource(
+    waiter: str,
+    resource: dict[str, str],
+    start_msg: str,
+    timeout_msg: str,
+    wait_mins: int = 60,
+):
+    click.echo(start_msg, nl=False)
+    waiter = rds_client.get_waiter(waiter)
+    finished = False
+    for _ in range(wait_mins):
+        try:
+            waiter.wait(
+                WaiterConfig=WaiterConfigTypeDef(Delay=60, MaxAttempts=2), **resource
+            )
+            finished = True
+            break
+        except WaiterError:
+            click.echo(".", nl=False)
+    click.echo("")
+    if not finished:
+        raise TimeoutError(timeout_msg)
+
+
+def get_latest_snapshot(rds_cluster_id: str) -> DBClusterSnapshotTypeDef:
+    page_iterator = rds_client.get_paginator("describe_db_cluster_snapshots").paginate(
+        DBClusterIdentifier=rds_cluster_id, SnapshotType="automated"
+    )
+    return sorted(
+        itertools.chain.from_iterable(
+            map(lambda page: page["DBClusterSnapshots"], page_iterator)
+        ),
+        key=lambda snapshot: snapshot["SnapshotCreateTime"],
+    )[-1]
+
+
+def restore_snapshot(snapshot: DBClusterSnapshotTypeDef) -> DBClusterTypeDef:
+    # Describe original cluster
+    cluster = rds_client.describe_db_clusters(
+        DBClusterIdentifier=snapshot["DBClusterIdentifier"]
+    )["DBClusters"][0]
+
+    # Restore snapshot to new cluster
+    restored_cluster = rds_client.restore_db_cluster_from_snapshot(
+        AvailabilityZones=snapshot["AvailabilityZones"],
+        DBClusterIdentifier=snapshot["DBClusterSnapshotIdentifier"].removeprefix(
+            "rds:"
+        ),
+        SnapshotIdentifier=snapshot["DBClusterSnapshotIdentifier"],
+        Engine=snapshot["Engine"],
+        EngineVersion=snapshot["EngineVersion"],
+        DBSubnetGroupName=cluster["DBSubnetGroup"],
+        DatabaseName=cluster["DatabaseName"],
+        VpcSecurityGroupIds=[
+            sg["VpcSecurityGroupId"] for sg in cluster["VpcSecurityGroups"]
+        ],
+        Tags=snapshot["TagList"],
+        EngineMode="provisioned",
+        DBClusterParameterGroupName=cluster["DBClusterParameterGroup"],
+        CopyTagsToSnapshot=True,
+        PubliclyAccessible=False,
+        ServerlessV2ScalingConfiguration=ServerlessV2ScalingConfigurationTypeDef(
+            MinCapacity=0.5, MaxCapacity=settings.rds_instance_acu
+        ),
+    )["DBCluster"]
+
+    # Wait until new cluster is active
+    wait_resource(
+        "db_cluster_available",
+        {"DBClusterIdentifier": restored_cluster["DBClusterIdentifier"]},
+        "Restoring to temporary cluster",
+        "Timed out when restoring cluster",
+    )
+
+    # Disable AutoMinorVersionUpgrade and set PreferredBackupWindow
+    restored_cluster = rds_client.modify_db_cluster(
+        DBClusterIdentifier=restored_cluster["DBClusterIdentifier"],
+        AutoMinorVersionUpgrade=False,
+        BackupRetentionPeriod=1,
+        PreferredBackupWindow="22:00-22:30",
```

Comment:
Why does this need to be set?

Reply:
If not set, there is a possibility that RDS will execute automatic backup just after the temporary cluster is created, prolonging the wait time (this happened to me once).
Unfortunately RDS aurora does not support disabling automatic backup, so I set it to a time outside [the usual backup window](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.Managing.Backups.html#Aurora.Managing.Backups.BackupWindow).